msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java

```diff
@@ -100,99 +115,39 @@ protected void onCreate(final Bundle savedInstanceState) {
         Telemetry.getInstance().startEvent(mTelemetryRequestId, mUiEventBuilder);
     }
 
+    private void launchWebView() {
+        if (mUseEmbeddedWebView) {
+            mChallengeCompletionCallback = new ChallengeCompletionCallback();
+            AzureActiveDirectoryWebViewClient webViewClient = new AzureActiveDirectoryWebViewClient(this, mAuthorizationRequest, mChallengeCompletionCallback);
+
+            setContentView(R.layout.activity_authentication);
+            final WebView webview = (WebView) this.findViewById(R.id.webview);
+            try {
+                mEmbeddedWebViewAuthorizationStrategy = new EmbeddedWebViewAuthorizationStrategy<>(webViewClient, webview);
+            } catch (final ClientException exception) {
+                sendError(exception.getErrorCode(), exception.getMessage());
+            } catch (final UnsupportedEncodingException exception) {
+                sendError(ErrorStrings.UNSUPPORTED_ENCODING, exception.getMessage());
+            }
+
+        } else {
+            mChromeCustomTabManager = new MsalChromeCustomTabManager(this);
+        }
+    }
+
     @Override
     protected void onStart() {
         super.onStart();
-        if (mChromePackageWithCustomTabSupport != null) {
-            warmUpCustomTabs();
+        if (!mUseEmbeddedWebView) {
+            mChromeCustomTabManager.bindCustomTabsService();
```

Comment:
can we revisit this pattern of if (!mUseEmbeddedWebView), as the bindCustomTabsService has its own checks, so we can potentially get into it and then have no way to fallback or recover

Reply:
Addressed. Please have a look. @addev-ashish